plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx

```diff
@@ -91,8 +153,31 @@ const ProductCollectionContent = (
 		return null;
 	}
 
+	// Let's not render anything until default attributes are set.
+	if ( ! isInitialAttributesSet.current ) {
+		return null;
+	}
```

Comment:
Nice one! I think we should adopt a similar mechanism for `location` in case it needs to invoke `getEntityRecords` in which two re-renders happen and because of that two calls.

Reply:
Do you mean changing this condition to the following?
```tsx
if ( ! isInitialAttributesSet.current & !location) {
```

> `location` in case it needs to invoke `getEntityRecords`

I don't think this is the case now. Do you mean in future we may call `getEntityRecords` in `useGetLocation` hook? Am I missing something? 🤷🏻‍♂️